libres/tests/enkf/enkf_obs_paths_detailed.cpp

```diff
@@ -0,0 +1,179 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include <string>
+#include <iostream>
+
+#include <catch2/catch.hpp>
+
+#include <ert/util/test_work_area.hpp>
+#include <ert/util/test_util.h>
+
+#include <ert/enkf/enkf_obs.hpp>
+#include "../tmpdir.hpp"
+
+/*
+ * Set up workarea, create and write start of file.
+ * Returns pointer to workares since it needs to be kept alive
+ * for the actual test
+ */
+void setup() {
+    util_make_path("obs_path");
+    FILE *stream = fopen("obs_path/conf.txt", "w");
+    fprintf(stream, "GENERAL_OBSERVATION WPR_DIFF_1 {"
+                    "DATA       = SNAKE_OIL_WPR_DIFF;"
+                    "INDEX_LIST = 400,800,1200,1800;"
+                    "RESTART    = 199;");
+    fclose(stream);
+}
+
+/*
+ * Append keywords and write closing-bracket.
+ * Returns pointer to the parsed file
+ */
+conf_instance_type *write_conf(std::vector<std::string> keyord_lines) {
+    FILE *stream = fopen("obs_path/conf.txt", "a");
+    for (auto s : keyord_lines)
+        fprintf(stream, "%s", s.data());
+    fprintf(stream, "};");
+    fclose(stream);
+
+    conf_class_type *enkf_conf_class = enkf_obs_get_obs_conf_class();
+    return conf_instance_alloc_from_file(enkf_conf_class, "enkf_conf",
+                                         "obs_path/conf.txt");
+}
+
+/*
+ * Just a convenient wrapper from char*  to std::string
+ */
+std::string get_path_error(conf_instance_type *enkf_conf) {
+    char *errors = conf_instance_get_path_error(enkf_conf);
+    if (errors) {
+        std::string s(errors);
+        free(errors);
+        return s;
+    } else
+        return std::string("");
+}
+
+/*
+ * Syntactic sugar for checking if a string contains another string
+ * The memmber-function ::contains() is introduced in C++23
+ * */
+bool contains(std::string errors, std::string kw) {
+    return errors.find(kw) != std::string::npos;
+}
+
+void touch_file(std::string name) {
+    FILE *stream = fopen(name.data(), "w");
+    fclose(stream);
+}
+
+TEST_CASE("Test parsing keywords in configuration", "[unittest]") {
+    GIVEN("A workearea") {
+        WITH_TMPDIR;
+        setup();
+
```

Comment:
Maybe since setup is just used here we should simply inline it instead of having a function with the overly generic name `setup`.